src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/NoClip.kt

```diff
@@ -6,37 +6,59 @@
 package net.ccbluex.liquidbounce.features.module.modules.movement
 
 import net.ccbluex.liquidbounce.event.EventTarget
-import net.ccbluex.liquidbounce.event.UpdateEvent
+import net.ccbluex.liquidbounce.event.MoveEvent
 import net.ccbluex.liquidbounce.features.module.Module
 import net.ccbluex.liquidbounce.features.module.ModuleCategory
+import net.ccbluex.liquidbounce.utils.MovementUtils.direction
+import net.ccbluex.liquidbounce.utils.MovementUtils.isMoving
+import net.ccbluex.liquidbounce.value.FloatValue
+import kotlin.math.cos
+import kotlin.math.sin
 
 object NoClip : Module("NoClip", ModuleCategory.MOVEMENT) {
+    val speed by FloatValue("Speed", 0.5f, 0f..10f)
 
     override fun onDisable() {
         mc.thePlayer?.noClip = false
     }
 
     @EventTarget
-    fun onUpdate(event: UpdateEvent) {
+    fun onMove(event: MoveEvent) {
         val thePlayer = mc.thePlayer ?: return
 
+		if (isMoving) {
+			val dir = direction;
+			
+			val xDir = -sin(dir) * speed
+			val zDir = cos(dir) * speed
+			
+			event.x = xDir
+			event.z = zDir
+			thePlayer.motionX = xDir * 0.9
+			thePlayer.motionZ = zDir * 0.9
+		}
+		else
+		{
+			event.x = 0.0
+			event.z = 0.0
+			thePlayer.motionX = 0.0
+			thePlayer.motionZ = 0.0
```

Comment:
> Why not just MovementUtils.strafe(speed)? Don't even have to check for movement input or calculate anything that way.
> 
> Should have also used: event.zeroXZ() thePlayer.stopXZ()

strafe sets motion afaik, which is turbo giga bad as it is affected by friction (speed differs when changing direction, goes up and down when changing directions instead of being 100% steady)